eng/tools/TypeSpecValidation/src/rules/folder-structure.ts

```diff
@@ -18,6 +20,37 @@ export class FolderStructureRule implements Rule {
       }
     });
 
+    // Verify top level folder is lower case
+    let folderStruct = folder.split(path.sep)
+    if (folderStruct[1].match(/[A-Z]/g)) {
+      success = false;
+      errorOutput += `Invalid folder name. Folders under specification/ must be lower case.\n`
+    }
+    
+    // Verify second level folder is capitalized after each '.'
```

Comment:
Update to support specs at either 2 or 3 levels deep (but not shallower or deeper) below `specification`:

```
# Valid
specification/codesigning/CodeSigning/tspconfig.yaml
specification/cognitiveservices/HealthInsights/HealthInsights.OpenAPI/tspconfig.yaml

# Invalid
specification/FooBar/tspconfig.yaml
specification/foo/bar/baz/Qux/tspconfig.yaml
```